DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Centre/Administrator/CentreAdministratorsViewModel.cs

```diff
@@ -8,26 +8,54 @@
 
     public class CentreAdministratorsViewModel : BaseSearchablePageViewModel
     {
+        private static readonly IEnumerable<(string, string)> RoleOptions = new[]
+        {
+            AdminFilterOptions.CentreAdministrator,
+            AdminFilterOptions.Supervisor,
+            AdminFilterOptions.Trainer,
+            AdminFilterOptions.ContentCreatorLicense,
+            AdminFilterOptions.CmsAdministrator,
+            AdminFilterOptions.CmsManager
+        };
+
+        private static readonly IEnumerable<(string, string)> AccountStatusOptions = new[]
+        {
+            AdminFilterOptions.IsLocked,
+            AdminFilterOptions.IsNotLocked
+        };
+
         public CentreAdministratorsViewModel(
             int centreId,
             IEnumerable<AdminUser> adminUsers,
+            IEnumerable<string> categories,
             string? searchString,
             string sortBy,
             string sortDirection,
+            string? filterString,
             int page
-        ) : base(searchString, sortBy, sortDirection, page)
+        ) : base(searchString, sortBy, sortDirection, filterString, page)
         {
             CentreId = centreId;
             var sortedItems = GenericSortingHelper.SortAllItems(
                 adminUsers.AsQueryable(),
                 sortBy,
                 sortDirection
             );
-            var searchedItems = GenericSearchHelper.SearchItems(sortedItems, SearchString).ToList();
-            MatchingSearchResults = searchedItems.Count;
+            var searchedItems = GenericSearchHelper.SearchItems(sortedItems, SearchString);
+            var filteredItems = FilteringHelper.FilterItems(searchedItems.AsQueryable(), filterString).ToList();
+            MatchingSearchResults = filteredItems.Count;
             SetTotalPages();
-            var paginatedItems = GetItemsOnCurrentPage(searchedItems);
+            var paginatedItems = GetItemsOnCurrentPage(filteredItems);
             Admins = paginatedItems.Select(adminUser => new SearchableAdminViewModel(adminUser));
+            IEnumerable<(string, string)> categoryOptions =
+                categories.Select(c => (c, $"{nameof(AdminUser.CategoryName)}|{c}"));
+
+            Filters = new[]
+            {
+                new FilterViewModel("Role", "Role", RoleOptions),
+                new FilterViewModel(nameof(AdminUser.CategoryName), "Category", categoryOptions),

```

Comment:
It's not used anywhere yet at least. Will change it to just be a string for now, but might bring it back for the JS version. Role isn't a nameof because there is no singular Role property to use.